### PR TITLE
[rfc] [tune/rllib] Fetch _progress_metrics from trainable for verbose=2 display

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

RLLibs trainables produce a large number of metrics which makethe log output with verbose=2 illegible. This PR introduces a private `_progress_metrics` property for trainables. If set, the trial progress callback will only print these metrics per default, unless overridden e.g. with a custom `TrialProgressCallback`.

Before:

`Trial PPO_CartPole-v0_d7d4d_00000 reported custom_metrics={},episode_media={},info={'learner': {'default_policy': {'learner_stats': {'cur_kl_coeff': 0.20000000298023224, 'cur_lr': 4.999999873689376e-05, 'total_loss': 8.785832, 'policy_loss': -0.033613548, 'vf_loss': 8.814276, 'vf_explained_var': 0.0020746698, 'kl': 0.025849672, 'entropy': 0.66827893, 'entropy_coeff': 0.0, 'model': {}}, 'custom_metrics': {}, 'num_agent_steps_trained': 128.0}}, 'num_env_steps_sampled': 4000, 'num_env_steps_trained': 4000, 'num_agent_steps_sampled': 4000, 'num_agent_steps_trained': 4000},sampler_results={'episode_reward_max': 69.0, 'episode_reward_min': 9.0, 'episode_reward_mean': 21.91160220994475, 'episode_len_mean': 21.91160220994475, 'episode_media': {}, 'episodes_this_iter': 181, 'policy_reward_min': {}, 'policy_reward_max': {}, 'policy_reward_mean': {}, 'custom_metrics': {}, 'hist_stats': {'episode_reward': [40.0, 31.0, 21.0, 13.0, 17.0, 15.0, 17.0, 15.0, 19.0, 25.0, 59.0, 25.0, 10.0, 30.0, 28.0, 20.0, 30.0, 16.0, 16.0, 23.0, 34.0, 31.0, 23.0, 11.0, 24.0, 31.0, 19.0, 24.0, 13.0, 19.0, 15.0, 18.0, 13.0, 35.0, 47.0, 16.0, 18.0, 29.0, 20.0, 27.0, 19.0, 41.0, 12.0, 15.0, 12.0, 27.0, 17.0, 18.0, 13.0, 69.0, 9.0, 35.0, 11.0, 9.0, 24.0, 22.0, 25.0, 25.0, 11.0, 36.0, 13.0, 56.0, 18.0, 11.0, 16.0, 14.0, 16.0, 22.0, 20.0, 30.0, 17.0, 11.0, 17.0, 32.0, 18.0, 11.0, 17.0, 10.0, 18.0, 29.0, 12.0, 29.0, 27.0, 34.0, 24.0, 21.0, 15.0, 12.0, 11.0, 13.0, 34.0, 21.0, 14.0, 20.0, 25.0, 17.0, 15.0, 13.0, 18.0, 20.0, 18.0, 35.0, 17.0, 14.0, 10.0, 16.0, 22.0, 12.0, 23.0, 19.0, 19.0, 25.0, 13.0, 15.0, 15.0, 13.0, 21.0, 20.0, 36.0, 65.0, 24.0, 15.0, 22.0, 26.0, 12.0, 14.0, 20.0, 23.0, 35.0, 19.0, 17.0, 19.0, 19.0, 15.0, 9.0, 21.0, 30.0, 18.0, 34.0, 13.0, 22.0, 28.0, 23.0, 14.0, 18.0, 15.0, 34.0, 24.0, 11.0, 15.0, 12.0, 28.0, 12.0, 13.0, 31.0, 19.0, 15.0, 37.0, 20.0, 32.0, 31.0, 20.0, 13.0, 10.0, 17.0, 31.0, 68.0, 19.0, 19.0, 23.0, 23.0, 18.0, 28.0, 20.0, 13.0, 26.0, 42.0, 20.0, 36.0, 14.0, 41.0], 'episode_lengths': [40, 31, 21, 13, 17, 15, 17, 15, 19, 25, 59, 25, 10, 30, 28, 20, 30, 16, 16, 23, 34, 31, 23, 11, 24, 31, 19, 24, 13, 19, 15, 18, 13, 35, 47, 16, 18, 29, 20, 27, 19, 41, 12, 15, 12, 27, 17, 18, 13, 69, 9, 35, 11, 9, 24, 22, 25, 25, 11, 36, 13, 56, 18, 11, 16, 14, 16, 22, 20, 30, 17, 11, 17, 32, 18, 11, 17, 10, 18, 29, 12, 29, 27, 34, 24, 21, 15, 12, 11, 13, 34, 21, 14, 20, 25, 17, 15, 13, 18, 20, 18, 35, 17, 14, 10, 16, 22, 12, 23, 19, 19, 25, 13, 15, 15, 13, 21, 20, 36, 65, 24, 15, 22, 26, 12, 14, 20, 23, 35, 19, 17, 19, 19, 15, 9, 21, 30, 18, 34, 13, 22, 28, 23, 14, 18, 15, 34, 24, 11, 15, 12, 28, 12, 13, 31, 19, 15, 37, 20, 32, 31, 20, 13, 10, 17, 31, 68, 19, 19, 23, 23, 18, 28, 20, 13, 26, 42, 20, 36, 14, 41]}, 'sampler_perf': {'mean_raw_obs_processing_ms': 0.10830351748146258, 'mean_inference_ms': 0.6018963459042099, 'mean_action_processing_ms': 0.07218363826300629, 'mean_env_wait_ms': 0.0565870985982077, 'mean_env_render_ms': 0.0}},episode_reward_max=69.0,episode_reward_min=9.0,episode_reward_mean=21.91160220994475,episode_len_mean=21.91160220994475,episodes_this_iter=181,policy_reward_min={},policy_reward_max={},policy_reward_mean={},hist_stats={'episode_reward': [40.0, 31.0, 21.0, 13.0, 17.0, 15.0, 17.0, 15.0, 19.0, 25.0, 59.0, 25.0, 10.0, 30.0, 28.0, 20.0, 30.0, 16.0, 16.0, 23.0, 34.0, 31.0, 23.0, 11.0, 24.0, 31.0, 19.0, 24.0, 13.0, 19.0, 15.0, 18.0, 13.0, 35.0, 47.0, 16.0, 18.0, 29.0, 20.0, 27.0, 19.0, 41.0, 12.0, 15.0, 12.0, 27.0, 17.0, 18.0, 13.0, 69.0, 9.0, 35.0, 11.0, 9.0, 24.0, 22.0, 25.0, 25.0, 11.0, 36.0, 13.0, 56.0, 18.0, 11.0, 16.0, 14.0, 16.0, 22.0, 20.0, 30.0, 17.0, 11.0, 17.0, 32.0, 18.0, 11.0, 17.0, 10.0, 18.0, 29.0, 12.0, 29.0, 27.0, 34.0, 24.0, 21.0, 15.0, 12.0, 11.0, 13.0, 34.0, 21.0, 14.0, 20.0, 25.0, 17.0, 15.0, 13.0, 18.0, 20.0, 18.0, 35.0, 17.0, 14.0, 10.0, 16.0, 22.0, 12.0, 23.0, 19.0, 19.0, 25.0, 13.0, 15.0, 15.0, 13.0, 21.0, 20.0, 36.0, 65.0, 24.0, 15.0, 22.0, 26.0, 12.0, 14.0, 20.0, 23.0, 35.0, 19.0, 17.0, 19.0, 19.0, 15.0, 9.0, 21.0, 30.0, 18.0, 34.0, 13.0, 22.0, 28.0, 23.0, 14.0, 18.0, 15.0, 34.0, 24.0, 11.0, 15.0, 12.0, 28.0, 12.0, 13.0, 31.0, 19.0, 15.0, 37.0, 20.0, 32.0, 31.0, 20.0, 13.0, 10.0, 17.0, 31.0, 68.0, 19.0, 19.0, 23.0, 23.0, 18.0, 28.0, 20.0, 13.0, 26.0, 42.0, 20.0, 36.0, 14.0, 41.0], 'episode_lengths': [40, 31, 21, 13, 17, 15, 17, 15, 19, 25, 59, 25, 10, 30, 28, 20, 30, 16, 16, 23, 34, 31, 23, 11, 24, 31, 19, 24, 13, 19, 15, 18, 13, 35, 47, 16, 18, 29, 20, 27, 19, 41, 12, 15, 12, 27, 17, 18, 13, 69, 9, 35, 11, 9, 24, 22, 25, 25, 11, 36, 13, 56, 18, 11, 16, 14, 16, 22, 20, 30, 17, 11, 17, 32, 18, 11, 17, 10, 18, 29, 12, 29, 27, 34, 24, 21, 15, 12, 11, 13, 34, 21, 14, 20, 25, 17, 15, 13, 18, 20, 18, 35, 17, 14, 10, 16, 22, 12, 23, 19, 19, 25, 13, 15, 15, 13, 21, 20, 36, 65, 24, 15, 22, 26, 12, 14, 20, 23, 35, 19, 17, 19, 19, 15, 9, 21, 30, 18, 34, 13, 22, 28, 23, 14, 18, 15, 34, 24, 11, 15, 12, 28, 12, 13, 31, 19, 15, 37, 20, 32, 31, 20, 13, 10, 17, 31, 68, 19, 19, 23, 23, 18, 28, 20, 13, 26, 42, 20, 36, 14, 41]},sampler_perf={'mean_raw_obs_processing_ms': 0.10830351748146258, 'mean_inference_ms': 0.6018963459042099, 'mean_action_processing_ms': 0.07218363826300629, 'mean_env_wait_ms': 0.0565870985982077, 'mean_env_render_ms': 0.0},num_healthy_workers=2,num_agent_steps_sampled=4000,num_agent_steps_trained=4000,num_env_steps_sampled=4000,num_env_steps_trained=4000,num_env_steps_sampled_this_iter=4000,num_env_steps_trained_this_iter=4000,num_steps_trained_this_iter=4000,agent_timesteps_total=4000,timers={'training_iteration_time_ms': 3653.196, 'load_time_ms': 0.194, 'load_throughput': 20610830.467, 'learn_time_ms': 1926.183, 'learn_throughput': 2076.646, 'synch_weights_time_ms': 2.433},counters={'num_env_steps_sampled': 4000, 'num_env_steps_trained': 4000, 'num_agent_steps_sampled': 4000, 'num_agent_steps_trained': 4000},perf={'cpu_util_percent': 23.366666666666664, 'ram_util_percent': 71.18333333333334} with parameters={'env': 'CartPole-v0'}.`

After:

`Trial PPO_CartPole-v0_00705_00000 reported num_env_steps_sampled=40000,num_env_steps_trained=40000,episode_reward_mean=192.96 with parameters={'env': 'CartPole-v0'}. This trial completed.`

## Related issue number
https://discuss.ray.io/t/tune-log-spam-from-rllib-trials/6047/6
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
Can you run an rllib script with this change 

Let's say rllib train -f rLlib/tuned_examples/appo/cartpole-appo.yaml and then post the output here? 

(The path to that yaml is something I guesstimated I can get you something more absolute if you need it.)

File: python/ray/tune/progress_reporter.py [hunk 1]
```diff
@@ -52,6 +57,21 @@
...
+    "hist_stats",
+    "sampler_results",
+    "sampler_perf",
+    "info",
+    "timers",
+    "perf",
+    "counters",
+    "episode_media",
+    "num_healthy_workers",
+}
```
Comment (richardliaw) [inline python/ray/tune/progress_reporter.py:72]: This should remain in the RLlib module

File: rllib/algorithms/algorithm.py [hunk 2]
```diff
@@ -198,6 +198,13 @@ class Algorithm(Trainable):
     # List of keys that are always fully overridden if present in any dict or sub-dict
     _override_all_key_list = ["off_policy_estimation_methods"]
 
+    _progress_metrics = [
```
Comment (richardliaw) [inline rllib/algorithms/algorithm.py:201]: since this is referenced outside the module, could we make it public?

**Review by richardliaw — DISMISSED** [review]

**Comment by gjoliver** [comment]
should we just beautify the result dict output?
I feel like the point of --verbose=2 is to see a lot of things that you normally don't see ...

File: python/ray/tune/progress_reporter.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/progress_reporter.py:72]: Ah this is a leftover from a previous iteration, I'll remove this

**Comment by krfricke** [comment]
@gjoliver we still ahve verbose=3, which is the default, and will print all metrics in a nice table. verbose=2 was introduced specifically for short and less-verbose outputs

File: rllib/algorithms/algorithm.py (hunk 2 above)
Comment (krfricke) [reply rllib/algorithms/algorithm.py:201]: I'd like to keep it internal for now as we will want to refactor all of progress reporting soon (see also https://github.com/ray-project/ray/issues/24604)

I don't want to rush out a public seeming API before we design that. Shall we keep this internal for now and consider promoting once we tackle the reporting?

**Comment by krfricke** [comment]
@gjoliver the default output (verbose=3) is this:

```
Result for PPO_CartPole-v0_ab738_00000:
  agent_timesteps_total: 8000
  counters:
    num_agent_steps_sampled: 8000
    num_agent_steps_trained: 8000
    num_env_steps_sampled: 8000
    num_env_steps_trained: 8000
  custom_metrics: {}
  date: 2022-07-26_10-17-14
  done: true
  episode_len_mean: 41.77
  episode_media: {}
  episode_reward_max: 121.0
  episode_reward_mean: 41.77
  episode_reward_min: 10.0
  episodes_this_iter: 90
  episodes_total: 278
  experiment_id: b4eb9f0a3b4b4c00bbd60bacadfea947
  hostname: Kais-MacBook-Pro.local
  info:
    learner:
      default_policy:
        custom_metrics: {}
        learner_stats:
          cur_kl_coeff: 0.30000001192092896
          cur_lr: 4.999999873689376e-05
          entropy: 0.6102302670478821
          entropy_coeff: 0.0
          kl: 0.01773761957883835
          model: {}
          policy_loss: -0.0348038375377655
          total_loss: 9.346200942993164
          vf_explained_var: -0.0739881619811058
          vf_loss: 9.37568473815918
        num_agent_steps_trained: 128.0
    num_agent_steps_sampled: 8000
    num_agent_steps_trained: 8000
    num_env_steps_sampled: 8000
    num_env_steps_trained: 8000
  iterations_since_restore: 2
  node_ip: 127.0.0.1
  num_agent_steps_sampled: 8000
  num_agent_steps_trained: 8000
  num_env_steps_sampled: 8000
  num_env_steps_sampled_this_iter: 4000
  num_env_steps_trained: 8000
  num_env_steps_trained_this_iter: 4000
  num_healthy_workers: 2
  num_steps_trained_this_iter: 4000
  perf:
    cpu_util_percent: 34.56666666666667
    ram_util_percent: 68.85000000000001
  pid: 98939
  policy_reward_max: {}
  policy_reward_mean: {}
  policy_reward_min: {}
  sampler_perf:
    mean_action_processing_ms: 0.07762884208869066
    mean_env_render_ms: 0.0
    mean_env_wait_ms: 0.061538921898287864
    mean_inference_ms: 0.6284820412361133
    mean_raw_obs_processing_ms: 0.11467544446062762
  sampler_results:
    custom_metrics: {}
    episode_len_mean: 41.77
    episode_media: {}
    episode_reward_max: 121.0
    episode_reward_mean: 41.77
    episode_reward_min: 10.0
    episodes_this_iter: 90
    hist_stats:
      episode_lengths:
      - 11
      - 17
      - 23
      - 23
      - 14
      - 45
      - 22
      - 17
      - 21
      - 19
      - 33
      - 71
      - 58
      - 31
      - 78
      - 39
      - 49
      - 32
      - 39
      - 27
      - 50
      - 53
      - 42
      - 108
      - 41
      - 73
      - 16
      - 47
      - 55
      - 44
      - 21
      - 31
      - 16
      - 32
      - 32
      - 11
      - 60
      - 30
      - 57
      - 17
      - 69
      - 43
      - 27
      - 31
      - 45
      - 55
      - 82
      - 16
      - 31
      - 41
      - 114
      - 26
      - 23
      - 26
      - 37
      - 29
      - 36
      - 11
      - 36
      - 29
      - 28
      - 45
      - 100
      - 16
      - 38
      - 23
      - 23
      - 78
      - 24
      - 121
      - 17
      - 72
      - 69
      - 22
      - 93
      - 19
      - 72
      - 94
      - 43
      - 14
      - 12
      - 31
      - 28
      - 59
      - 27
      - 76
      - 36
      - 60
      - 33
      - 27
      - 10
      - 66
      - 11
      - 54
      - 51
      - 57
      - 115
      - 43
      - 38
      - 20
      episode_reward:
      - 11.0
      - 17.0
      - 23.0
      - 23.0
      - 14.0
      - 45.0
      - 22.0
      - 17.0
      - 21.0
      - 19.0
      - 33.0
      - 71.0
      - 58.0
      - 31.0
      - 78.0
      - 39.0
      - 49.0
      - 32.0
      - 39.0
      - 27.0
      - 50.0
      - 53.0
      - 42.0
      - 108.0
      - 41.0
      - 73.0
      - 16.0
      - 47.0
      - 55.0
      - 44.0
      - 21.0
      - 31.0
      - 16.0
      - 32.0
      - 32.0
      - 11.0
      - 60.0
      - 30.0
      - 57.0
      - 17.0
      - 69.0
      - 43.0
      - 27.0
      - 31.0
      - 45.0
      - 55.0
      - 82.0
      - 16.0
      - 31.0
      - 41.0
      - 114.0
      - 26.0
      - 23.0
      - 26.0
      - 37.0
      - 29.0
      - 36.0
      - 11.0
      - 36.0
      - 29.0
      - 28.0
      - 45.0
      - 100.0
      - 16.0
      - 38.0
      - 23.0
      - 23.0
      - 78.0
      - 24.0
      - 121.0
      - 17.0
      - 72.0
      - 69.0
      - 22.0
      - 93.0
      - 19.0
      - 72.0
      - 94.0
      - 43.0
      - 14.0
      - 12.0
      - 31.0
      - 28.0
      - 59.0
      - 27.0
      - 76.0
      - 36.0
      - 60.0
      - 33.0
      - 27.0
      - 10.0
      - 66.0
      - 11.0
      - 54.0
      - 51.0
      - 57.0
      - 115.0
      - 43.0
      - 38.0
      - 20.0
    policy_reward_max: {}
    policy_reward_mean: {}
    policy_reward_min: {}
    sampler_perf:
      mean_action_processing_ms: 0.07762884208869066
      mean_env_render_ms: 0.0
      mean_env_wait_ms: 0.061538921898287864
      mean_inference_ms: 0.6284820412361133
      mean_raw_obs_processing_ms: 0.11467544446062762
  time_since_restore: 7.894150733947754
  time_this_iter_s: 3.864184856414795
  time_total_s: 7.894150733947754
  timers:
    learn_throughput: 1874.24
    learn_time_ms: 2134.198
    load_throughput: 13859740.603
    load_time_ms: 0.289
    synch_weights_time_ms: 2.067
    training_iteration_time_ms: 3942.189
  timestamp: 1658827034
  timesteps_since_restore: 0
  timesteps_total: 8000
  training_iteration: 2
  trial_id: ab738_00000
  warmup_time: 10.877946138381958
```

While I personally think it's (much) too verbose, it includes all metrics and all information that could possibly be necessary for a user. 
Verbose=2 is specifically meant for short and legible outputs so should only contain the most important information. In the current state, verbose=2 is not usable at all for rllib, it contains much too much information and the relevant metrics cannot be found.

My proposal is:

- Merge this PR to fix verbose=2
- Tackle progress reporting in a general way after the branch cut
- This will include easy customizability for displayed metrics as well as sensible defaults

Please note that all metrics are exported for each step in csvs, so e.g. the single episode rewards can always be accessed. But having them clutter the log output is probably not great.

**Review by avnishn — APPROVED** [review]

**Review by gjoliver — APPROVED** [review]
can you add a comment about what verbose=2 looks like after this change?
I didn't even know there is verbose=3.

File: python/ray/tune/progress_reporter.py [hunk 3]
```diff
@@ -52,6 +57,9 @@
     IS_NOTEBOOK = False
 
 
+SKIP_RESULTS_IN_REPORT = {"config", TRIAL_ID, EXPERIMENT_TAG, DONE}
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/progress_reporter.py:60]: DONE is a metric that instructs tune to stop the trial. In verbose=2 this is reflected by the suffix string "This trial terminated." The progress table will always show the correct state

File: python/ray/tune/progress_reporter.py [hunk 4]
```diff
@@ -1179,16 +1194,27 @@ def log_result(self, trial: "Trial", result: Dict, error: bool = False):
...
-            print_result.pop(auto_result, None)
-
-        print_result_str = ",".join([f"{k}={v}" for k, v in print_result.items()])
+        if self._progress_metrics:
+            # If progress metrics are given, only report these
+            flat_result = flatten_dict(result)
+
+            print_result = {}
+            for metric in self._progress_metrics:
+                print_result[metric] = flat_result.get(metric)
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/progress_reporter.py:1203]: The left hand side is being set, so we need the [] syntax here. The right hand side is user-specified (the `metric`), so we should soft lookup. `.get()` returns `None` per default which is a good default value here

**Comment by krfricke** [comment]
Thanks - the example output for verbose=2 is in the PR description. Merging with 2 approving reviews

**Comment by vakker** [comment]
@krfricke And what if I don't want to keep the `Trial ... reported ...` messages at all, but keep the table? I have a lot of `parameters`, and these prints don't add any value (since there's the table anyway).
E.g. currently I'm getting this: 
```
== Status ==                                                                                                                                                                                                       
Current time: 2022-09-13 09:52:34 (running for 23:09:29.09)                                                                                                                                                        
Memory usage on this node: 35.1/125.5 GiBNone                                                                                                                                                                      
Using AsyncHyperBand: num_stopped=1                                                                                                                                                                                
Bracket: Iter 1000.000: -19.9275             
Resources requested: 10.999999999999982/20 CPUs, 2.0/2 GPUs, 0.0/76.91 GiB heap, 0.0/36.95 GiB objects (0.0/1.0 accelerator_type:G)
Result logdir: .../exp-set-03/exp-01/PPO.2022-09-12.10:43:05                                                     Number of trials: 10/10 (5 PENDING, 2 RUNNING, 3 TERMINATED)                                                                                                                                                       
+-----------------------------+------------+----------------------+--------+------------------+----------+----------+------------------------+----------------------+----------------------+
| Trial name                  | status     | loc                  |   iter |   total time (s) |       ts |   reward |   num_recreated_wor... |   episode_reward_max |   episode_reward_min |                       
|-----------------------------+------------+----------------------+--------+------------------+----------+----------+------------------------+----------------------+----------------------|
| trial-4d5be_00003-lr:0.0001 | RUNNING    | 10.2.213.186:823414  |   1143 |         22764.5  |  5915025 |   -19.99 |                      0 |                  -17 |                  -21 |                       | trial-4d5be_00004-lr:1e-05  | RUNNING    | 10.2.213.186:2215022 |    112 |          2257.54 |   579600 |   -20.25 |                      0 |                  -18 |                  -21 |
| trial-4d5be_00005-lr:0.1    | PENDING    |                      |        |                  |          |          |                        |                      |                      |                       | trial-4d5be_00006-lr:0.01   | PENDING    |                      |        |                  |          |          |                        |                      |                      |                       | trial-4d5be_00007-lr:0.001  | PENDING    |                      |        |                  |          |          |                        |                      |                      |
| trial-4d5be_00008-lr:0.0001 | PENDING    |                      |        |                  |          |          |                        |                      |                      |
| trial-4d5be_00009-lr:1e-05  | PENDING    |                      |        |                  |          |          |                        |                      |                      |
| trial-4d5be_00000-lr:0.1    | TERMINATED | 10.2.213.186:956546  |   1000 |         20413.7  |  5175000 |   -21    |                      0 |                  -21 |                  -21 |
| trial-4d5be_00001-lr:0.01   | TERMINATED | 10.2.213.186:958343  |   3000 |         60241.4  | 15525000 |   -20.14 |                      0 |                  -17 |                  -21 |
| trial-4d5be_00002-lr:0.001  | TERMINATED | 10.2.213.186:2322828 |   3000 |         60257.8  | 15525000 |   -20.23 |                      0 |                  -17 |                  -21 |
+-----------------------------+------------+----------------------+--------+------------------+----------+----------+------------------------+----------------------+----------------------+
                                                                                                                                                                                                                   
                                                                                                                                                                                                                   
Trial trial-4d5be_00004-lr:1e-05 reported num_env_steps_trained=584775,episode_reward_mean=-20.32,num_env_steps_sampled=584775 with parameters={'lambda': 0.95, 'kl_coeff': 0.5, 'clip_rewards': True, 'clip_param'
: 0.1, 'vf_clip_param': 10.0, 'entropy_coeff': 0.01, 'train_batch_size': 5000, 'rollout_fragment_length': 20, 'sgd_minibatch_size': 250, 'num_sgd_iter': 10, 'lr': 1e-05, 'vf_loss_coeff': 1.0, 'model': {'custom_m
odel': 'gnn_net', 'custom_model_config': {'graph_conv': 'GINFeatures', 'activation': 'LeakyReLU', 'dims': [32, 32, 32, 32, 32]}}, 'env': 'atari_graph', 'env_config': {'pg_name': 'PongNoFrameskip-v4', 'continuous
_actions': False, '__run': 'PPO', 'logdir': '.../exp-set-03/exp-01'}, 'num_workers': 45, 'evaluation_config': {'env_config
': {'is_eval': True}}, 'evaluation_interval': None, 'evaluation_num_episodes': 10, 'evaluation_duration_unit': 'episodes', 'num_cpus_per_worker': 0.1, 'evaluation_num_workers': 0, 'num_gpus': 1.0, 'framework': '
torch', 'num_envs_per_worker': 5, 'batch_mode': 'truncate_episodes', 'observation_filter': 'NoFilter', 'callbacks': <class 'spg_experiments.callbacks.CustomCallbacks'>}.        
```

**Pull request merged**